src/uct/base/uct_md.h

```diff
@@ -210,6 +210,75 @@ ucs_status_t uct_md_mem_alloc(uct_md_h md, size_t *length_p, void **address_p,
                               ucs_memory_type_t mem_type, unsigned flags,
                               const char *alloc_name, uct_mem_h *memh_p);
 
+
+/**
+ * @ingroup UCT_MD
+ * @brief Allocate memory for zero-copy sends and remote access.
+ *
+ * Allocate memory on the memory domain. In order to use this function, MD
+ * must support @ref UCT_MD_FLAG_ALLOC flag.
+ *
+ * @param [in]     md          Memory domain to allocate memory on.
+ * @param [in,out] params      Struct containing parameters such as memory type, address, allocation length, 
+ *                             and other parameters from @ref uct_md_mem_alloc function.
+ */
+
+typedef enum {
+    /** Enables @ref uct_md_mem_alloc_params_t::alloc_name */
+    UCT_MD_MEM_ALLOC_FIELD_ALLOC_NAME = UCS_BIT(0),
+    /** Enables @ref uct_md_mem_alloc_params_t::length */
+    UCT_MD_MEM_ALLOC_FIELD_LENGTH     = UCS_BIT(1),
+    /** Enables @ref uct_md_mem_alloc_params_t::mem_type */
+    UCT_MD_MEM_ALLOC_FIELD_MEM_TYPE   = UCS_BIT(2),
+    /** Enables @ref uct_md_mem_alloc_params_t::flags */
+    UCT_MD_MEM_ALLOC_FIELD_FLAGS      = UCS_BIT(3)
+} uct_md_mem_alloc_params_mask;
+
+typedef struct uct_md_mem_alloc_params {
+    /** 
+     *
+     */
+    uint64_t          field_mask;
+
+    /** 
+     * Name of the allocated region, used to track memory
+     * usage for debugging and profiling.
+     */
+    const char        *alloc_name;
+
+    /** 
+     * Filled with handle for allocated region.
+     */
+    uct_mem_h         *memh_p;
+
+    /**
+     * Points to the size of memory to allocate. Upon successful
+     * return, filled with the actual size that was allocated,
+     * which may be larger than the one requested. Must be >0.
+     */
+    size_t            *length_p;
+
+    /**
+     * The address
+     */
+    void              **address_p;
+
+    /**
+     * Memory type of the allocation
+     */
+    ucs_memory_type_t mem_type;
+
+    /**
+     * Memory allocation flags, see @ref uct_md_mem_flags
+     */
+    unsigned          flags;
+} uct_md_mem_alloc_params_t;
+
+typedef uct_md_mem_alloc_params_t* uct_md_mem_alloc_params_h;
```

Comment:
IMO no need for this one. `params` is not a "permanent" object accessed using a "handle" many times.